### ISSUE TITLE
datadog 5: fails to parse newer iostat output

### ISSUE THREAD

**Issue by bflyblue** [NixOS/nixpkgs#40103]
## Issue description

```
May 07 12:04:45 toolkit1 dd-agent[1791]: 2018-05-07 12:04:45,443 | ERROR | dd.collector | checks.collector(unix.py:220) | Cannot extract IO statistics
May 07 12:04:45 toolkit1 dd-agent[1791]: Traceback (most recent call last):
May 07 12:04:45 toolkit1 dd-agent[1791]:   File "/nix/store/raw9gp68al3nj6gg5mcqk32s8lm8ws2m-dd-agent-5.11.2/agent/checks/system/unix.py", line 139, in check
May 07 12:04:45 toolkit1 dd-agent[1791]:     io.update(self._parse_linux2(stdout))
May 07 12:04:45 toolkit1 dd-agent[1791]:   File "/nix/store/raw9gp68al3nj6gg5mcqk32s8lm8ws2m-dd-agent-5.11.2/agent/checks/system/unix.py", line 42, in _parse_linux2
May 07 12:04:45 toolkit1 dd-agent[1791]:     recentStats = output.split('Device:')[2].split('\n')
May 07 12:04:45 toolkit1 dd-agent[1791]: IndexError: list index out of range
May 07 12:04:45 toolkit1 dd.collector[1791]: ERROR (unix.py:220): Cannot extract IO statistics
                                             Traceback (most recent call last):
                                               File "/nix/store/raw9gp68al3nj6gg5mcqk32s8lm8ws2m-dd-agent-5.11.2/agent/checks/system/unix.py", line 139, in check
                                                 io.update(self._parse_linux2(stdout))
                                               File "/nix/store/raw9gp68al3nj6gg5mcqk32s8lm8ws2m-dd-agent-5.11.2/agent/checks/system/unix.py", line 42, in _parse_linux2
                                                 recentStats = output.split('Device:')[2].split('\n')
                                             IndexError: list index out of range
```

## Technical details

Newer versions of `iostat` have changed their output slightly and no longer include the colon after Device:

```
Device             tps    kB_read/s    kB_wrtn/s    kB_read    kB_wrtn
sda               0.92         0.51         9.95    1964789   38467824
```

Can confirm adding the following line to the `dd-agent` buildCommand does work around this issue:

```
sed -i 's/Device:/Device/' $out/agent/checks/system/unix.py
```

**Comment by Mic92** [comment]
Should be also reported upstream.

**Comment by bflyblue** [comment]
logged it upstream as well
https://github.com/DataDog/dd-agent/issues/3731

**Comment by tazjin** [comment]
As a fix for this issue (with backporting in mind), is it preferable to keep the current version of the agent while applying the workaround or should the version be bumped to a newer one that includes the fix?

The location of default config files seems to have changed in the dist that newer agent versions assemble, so an update may involve a little bit of work other than just bumping the source.

**Comment by bflyblue** [comment]
When I did a little poking around to investigate this issue it seemed like the version of the agent in Nix was a bit dated anyway, so my personal preference would be to use a newer version. Upstream patched their code 9 days ago according to the issue I linked to above.

**Comment by tazjin** [comment]
@bflyblue I'd like to have a fix that can be included in 18.03 without having to wait for 18.09 (we prefer to use as few packages from unstable in production), so I'm wondering if a version upgrade would be applied to 18.03 if it fixes an issue.

Maybe it makes sense to just do both. One PR (first) that adds a patch to the existing version (this should be backportable), and then a second one with a version bump.

**Comment by tazjin** [comment]
@bflyblue Submitted #41035 to fix this - version upgrade will be a separate PR.
